File: app/src/main/java/com/seafile/seadroid2/framework/data/model/user/UserModel.java
package com.seafile.seadroid2.framework.data.model.user;

import android.os.Parcel;
import android.os.Parcelable;

public class UserModel implements Parcelable {
    private String email;
    private String name;
    private String contact_email;
    private String avatar_url;

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getContactEmail() {
        return contact_email;
    }

    public void setContactEmail(String contact_email) {
        this.contact_email = contact_email;
    }

    public String getAvatarUrl() {
        return avatar_url;
    }

    public void setAvatarUrl(String avatar_url) {
        this.avatar_url = avatar_url;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.email);
        dest.writeString(this.name);
        dest.writeString(this.contact_email);
        dest.writeString(this.avatar_url);
    }

    public UserModel() {
    }

    protected UserModel(Parcel in) {
        this.email = in.readString();
        this.name = in.readString();
        this.contact_email = in.readString();
        this.avatar_url = in.readString();
    }

    public static final Parcelable.Creator<UserModel> CREATOR = new Parcelable.Creator<UserModel>() {
        @Override
        public UserModel createFromParcel(Parcel source) {
            return new UserModel(source);
        }

        @Override
        public UserModel[] newArray(int size) {
            return new UserModel[size];
        }
    };
}

File: app/src/main/java/com/seafile/seadroid2/framework/model/user/UserModel.kt
package com.seafile.seadroid2.framework.model.user

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

class UserModel() : Parcelable {

    var email: String? = null

    var name: String? = null

    @SerializedName("contact_email")
    private var contactEmailBacking: String? = null

    @SerializedName("avatar_url")
    private var avatarUrlBacking: String? = null

    fun getContactEmail(): String? = contactEmailBacking

    fun setContactEmail(contactEmail: String?) {
        contactEmailBacking = contactEmail
    }

    fun getAvatarUrl(): String? = avatarUrlBacking

    fun setAvatarUrl(avatarUrl: String?) {
        avatarUrlBacking = avatarUrl
    }

    override fun describeContents(): Int = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(email)
        dest.writeString(name)
        dest.writeString(contactEmailBacking)
        dest.writeString(avatarUrlBacking)
    }

    private constructor(parcel: Parcel) : this() {
        email = parcel.readString()
        name = parcel.readString()
        contactEmailBacking = parcel.readString()
        avatarUrlBacking = parcel.readString()
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<UserModel> = object : Parcelable.Creator<UserModel> {
            override fun createFromParcel(source: Parcel): UserModel = UserModel(source)
            override fun newArray(size: Int): Array<UserModel?> = arrayOfNulls(size)
        }
    }
}
